server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java

```diff
@@ -136,14 +136,18 @@ public static Query refExistsQuery(Reference ref, Context context, boolean count
             valueType = ArrayType.unnest(valueType);
         }
         StorageSupport<?> storageSupport = valueType.storageSupport();
-        if (storageSupport == null && ref instanceof DynamicReference) {
+        if (storageSupport == null) {
```

Comment:
To my understanding ref.valueType().storageSupport() should never be null. Will do more investigation.